http/httpclient/src/main/java/com/okta/commons/http/httpclient/HttpClientRequestExecutor.java

```diff
@@ -263,17 +260,114 @@ private HttpHeaders getHeaders(HttpResponse response) {
         return headers;
     }
 
-    private static int parseConfigValue(String key, int defaultValue, String warning) {
+    private int getRequestExecutorParam(String key, String warning, int defaultValue) {
+        String configuredValueString = this.requestExecutorParams.get(key);
+        try {
+            if (configuredValueString != null) {
+                return Integer.parseInt(configuredValueString);
+            }
+        } catch (NumberFormatException nfe) {
+            log.warn("{}: {}. Using default: {}.",
```

Comment:
nit: Improve error message to be something like
`Failed to parse configuration property [{key}: {value}] falling back to default value: {default}`

Reply:
Parsing logic moved into `lookupConfigValue` method. Outdated